lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -431,46 +431,6 @@ def test_enroll_staff_with_invalid_data(self):
         self.assertEqual(response.status_code, 302)
         self.assertRedirects(response, '/courses/{}/about'.format(unicode(self.course_key)))
 
-    def test_courseware_redirection(self):
-        """
-        Tests that a global staff member is redirected to the staff enrollment page.
-
-        Un-enrolled Staff user should be redirected to the staff enrollment page accessing courseware,
-        user chooses to enroll in the course. User is enrolled and redirected to the requested url.
-
-        Scenario:
-            1. Un-enrolled staff tries to access any course vertical (courseware url).
-            2. User is redirected to the staff enrollment page.
-            3. User chooses to enroll in the course.
-            4. User is enrolled in the course and redirected to the requested courseware url.
-        """
-        self._create_global_staff_user()
-        courseware_url, enroll_url = self._create_url_for_enroll_staff()
-
-        # Accessing the courseware url in which not enrolled & redirected to staff enrollment page
```

Comment:
Ah, I should have mentioned this. I have removed the special redirect logic for global staff. In all other situations, global staff are treated the same as course staff, who can enter a course as if they are enrolled. For some reason, the old courseware page had this special redirect (only for global staff) which required that they enroll. After speaking with @marcotuts, it seemed cleaner just to be consistent and allow global staff the same access as course staff. This has the nice benefit that global staff won't enroll in courses just to test them, which messes (slightly) with enrollment numbers.